zrml/swaps/rpc/src/lib.rs

```diff
@@ -35,6 +35,15 @@ where
         asset_out: Asset<MarketId>,
         at: Option<BlockHash>,
     ) -> Result<BalanceType>;
+
+    #[rpc(name = "swaps_getSpotPrices")]
+    fn get_spot_prices(
+        &self,
+        pool_id: PoolId,
+        asset_in: Asset<MarketId>,
+        asset_out: Asset<MarketId>,
+        blocks: Vec<BlockHash>,
+    ) -> Result<Vec<BalanceType>>;
```

Comment:
Differs from `get_spot_price` by its blocks inputs and balances output, i.e., good for historical data